libraries/core/src/test/kotlin/com/bumble/appyx/core/store/RetainedInstanceStoreTest.kt

```diff
@@ -0,0 +1,110 @@
+package com.bumble.appyx.core.store
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertSame
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class RetainedInstanceStoreTest {
+
+    private val identifier = "identifier"
+    private val store = RetainedInstanceStoreImpl()
+
+    @Test
+    fun `GIVEN object stored WHEN get with same identifier THEN same instance is retrieved`() {
+        val obj = Any()
+        store.get(identifier) { obj }
+
+        val retrieved = store.get(identifier) { Any() }
+
+        assertSame(obj, retrieved)
+    }
+
+    @Test
+    fun `GIVEN object stored WHEN get with same identifier THEN factory not called`() {
+        var factoryCalled = false
+        store.get(identifier) { Any() }
+
+        store.get(identifier) {
+            factoryCalled = true
+            Any()
+        }
+
+        assertFalse(factoryCalled)
```

Comment:
Same as below, we don't test the contract here

Reply:
fair enough. removed